### PR TITLE
initial commit; Featurs - login flow, aws cognito setup

### PR DESCRIPTION
**Pull request by Vivek205**
Please review the code and provide suggestions

### CODE REVIEW

File: src/components/AiMarketplace/MainSection/Filter/StyledExpansionPanel.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    };
+
+    render() {
+        const { classes } = this.props;
+        const { expansionData } = this.state;
+
+        return (
+            <Fragment>
+                {expansionData.map(val => (
+                    <ExpansionPanel className={classes.filterExpansionPanel}>
```
Comment (chetankothari) [inline src/components/AiMarketplace/MainSection/Filter/StyledExpansionPanel.js]: Key missing.

Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/Filter/StyledExpansionPanel.js]: fixed in the [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/09c13f188b76a84c0732bfb7f73cc9adadc102fa) and [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/930dfb3c8818b72defe2aa1e87f28cbfbe67f86b)

File: src/components/common/StyledButton/index.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+const StyledButton = props => {
+    const classes = useStyles();
+
+    return (
+        <Button
+            className={classNames(classes.styledButton, classes[buttonColor[props.type]])}
+            disabled={props.disabled}
+            onClick={props.onClick}
+        >
+            {props.iconClass ? <i className={props.iconClass}></i> : null}
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/common/StyledButton/index.js]: [commit ](https://github.com/singnet/refactored-dapp/pull/2/commits/bd712731e1117e0e5af937e5400c8895f2e53d35)

File: src/utility/Validation.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const isValidEmail = email => {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/utility/Validation.js]: validation will be raised as a separate PR after integrating ValidationJs

File: src/utility/stringConstants/APIEndpoints.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const APIEndpoints = {
+    GET_SERVICES_LIST: {
+        name: "Get Service",
+        endpoint: "https://3zjz5tihg2.execute-api.us-east-1.amazonaws.com/ropsten",
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/utility/stringConstants/APIEndpoints.js]: removed from the repo and added in .gitIgnore.
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/5a87a00ab775c1f2cfd2b4df0fe1c9cb675e69de)
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/84a0deed9734480b20ed41950076cd6f37b550a6)

File: src/utility/stringConstants/Session.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+const Session = {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/utility/stringConstants/Session.js]: renamed it.
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/4101b4cea1083a500549ed1081522a864093f222)
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/84a0deed9734480b20ed41950076cd6f37b550a6)

File: public/index.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="theme-color" content="#000000" />
+        <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+        <link
+            rel="stylesheet"
+            href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
+            integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
+            crossorigin="anonymous"
+        />
+        <title>React App</title>
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply public/index.html]: renamed it

File: public/manifest.json [hunk 7]
```diff
@@ -0,0 +1,15 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply public/manifest.json]: renamed it

File: src/App.js [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+const PageNotFound = lazy(() => import("./components/PageNotFound"));
+const AiMarketplace = lazy(() => import("./components/AiMarketplace"));
+const SignUp = lazy(() => import("./components/Login/Signup"));
+const Login = lazy(() => import("./components/Login"));
+
+Amplify.configure(aws_config);
+
+class App extends Component {
+    state = {
+        initialized: false,
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/App.js]: removed the state

File: src/Redux/actionCreators/ServiceActions.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import { Auth, API } from "aws-amplify";
+import { APIEndpoints } from "../../utility/stringConstants/APIEndpoints";
+
+export const FETCH_SERVICE_LIST = "SET_SERVICE_LIST";
+
+export const fetchService = ({ q, limit, offset, sort_by, order_by }) => (dispatch, getState) => {
+    const _q = typeof q !== "undefined" ? q : getState().serviceReducer.q;
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/ServiceActions.js]: renamed them to searchKeyword and working on that in the separate branch. **service-listing-remove-auth**

File: src/Redux/actionCreators/ServiceActions.js [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+                    });
+                })
+                .catch(() => {});
+        })
+        .catch(err => {
+            console.log(err);
+        });
+};
+
+// export const servicePagination = () => dispatch => {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/ServiceActions.js]: removed comments
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/02a37cf88cc7597ad1f3af414130c657af3aa4fd)
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/bfa283cc35581e820916f29559c32ab49baba81e)
[commit](https://github.com/singnet/refactored-dapp/pull/2/commits/9c2c6f00783543ecdeac19798a5afb8570910c61)

File: src/Redux/actionCreators/UserActions.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    };
+
+    return Auth.signIn(username, password)
+        .then(res => {
+            sessionStorage.setItem(Session.USERNAME, username);
+            userDetails.payload.isEmailVerified = res.attributes.email_verified;
+            userDetails.payload.isLoggedIn = true;
+        })
+        .catch(err => {
+            // if (err.code === "UserNotConfirmedException") {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/UserActions.js]: resolved [here](https://github.com/singnet/refactored-dapp/pull/2/files/1c984e410aaf281537f19c36d15377a1bd28dfb3#r299412772)

File: src/components/AiMarketplace/MainSection/ServiceCollection/StyledPagination/index.js [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import MenuItem from "@material-ui/core/MenuItem";
+import { connect } from "react-redux";
+
+import { useStyles } from "./styles";
+import { serviceActions } from "../../../../../Redux/actionCreators";
+
+const StyledPagination = ({ limit, offset, total_count, fetchService }) => {
+    const [itemsPerPage, setItemsPerPage] = useState(10);
+    // const [currentPage, setCurrentPage] = useState(1);
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/ServiceCollection/StyledPagination/index.js]: resolved [here](https://github.com/singnet/refactored-dapp/pull/2/files/1c984e410aaf281537f19c36d15377a1bd28dfb3#r299412772)

File: src/components/Login/Signup/styles.js [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+        fontSize: "12.17px",
+        letterSpacing: "0.4px",
+    },
+    passwordTxt: {
+        color: theme.palette.text.mediumShadeGray,
+        fontFamily: theme.typography.secondary.main,
+        fontSize: "12.17px",
+        letterSpacing: "0.4px",
+    },
+    // checkboxSection: {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Login/Signup/styles.js]: resolved [here](https://github.com/singnet/refactored-dapp/pull/2/files/1c984e410aaf281537f19c36d15377a1bd28dfb3#r299412772)

File: src/Redux/actionCreators/UserActions.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+            let path = `/wallet?username=${username}`;
+            let myInit = {
+                headers: { Authorization: currentSession.idToken.jwtToken },
+                queryStringParameters: {
+                    username,
+                },
+            };
+            API.get(apiName, path, myInit)
+                .then(res => {
+                    console.log("checkWalletStatus", res);
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/UserActions.js]: removed them [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/9c2c6f00783543ecdeac19798a5afb8570910c61)

File: src/Redux/actionCreators/UserActions.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+                    dispatch({
+                        type: CHECK_WALLET_STATUS,
+                        payload: { isWalletAssigned: res.data.isAssigned },
+                    });
+                })
+                .catch(() => {});
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/UserActions.js]: resolved them here [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/9c2c6f00783543ecdeac19798a5afb8570910c61)

File: src/Redux/reducers/ServiceReducer.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
+import { serviceActions } from "../actionCreators";
+
+const serviceList = {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/reducers/ServiceReducer.js]: renamed It

File: src/Redux/reducers/ServiceReducer.js [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    order_by: "desc",
+    total_count: 0,
+};
+
+const serviceReducer = (state = serviceList, action) => {
+    console.log("service Reducet", state, "payload", action.payload);
+    switch (action.type) {
+        case serviceActions.FETCH_SERVICE_LIST: {
+            let x = { ...state, ...action.payload };
+            console.log("aaaaaaa", x);
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/reducers/ServiceReducer.js]: removed them.
> in this [comment](https://github.com/singnet/refactored-dapp/pull/2/files/1c984e410aaf281537f19c36d15377a1bd28dfb3#r299413931)

File: src/assets/Theme.js [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    backgroundColor: {
+        blue: customBlue,
+        red: errorBgColor,
+        white: whiteColor,
+        githubBlack,
+    },
+});
+
+export default theme;
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/assets/Theme.js]: It is being continuously reduced based on the inputs from Sujith.

File: src/components/AiMarketplace/MainSection/ServiceCollection/StyledPagination/index.js [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+                        }
+                        className={classes.selectBox}
+                    >
+                        <MenuItem value={10}>Ten</MenuItem>
+                        <MenuItem value={20}>Twenty</MenuItem>
+                        <MenuItem value={30}>Thirty</MenuItem>
+                    </Select>
+                </FormControl>
+                <span>
+                    {offset}-{parseFloat(limit) + parseFloat(limit) * parseFloat(offset)} of {total_count}
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/ServiceCollection/StyledPagination/index.js]: extracted them to separate variable in [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/2272c9df86c078921c382c74d714f37ae395296b)

File: src/components/AiMarketplace/MainSection/ServiceCollection/ToolBar/SearchInputToggler.js [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+import clsx from "clsx";
+
+import { useStyles } from "./styles";
+
+const SearchInputToggler = ({ showSearchInput, toggleSearchInput, handleSearch, searchKeyword }) => {
+    const classes = useStyles();
+
+    return (
+        <Fragment>
+            {showSearchInput ? (
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/ServiceCollection/ToolBar/SearchInputToggler.js]: resolve it in [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/55cd7c47c1ebf2ed9802ca3d6637718136b9781d)

File: src/components/AiMarketplace/MainSection/ServiceCollection/ToolBar/index.js [hunk 21]
```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import Grid from "@material-ui/core/Grid";
+import Icon from "@material-ui/core/Icon";
+import clsx from "clsx";
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/ServiceCollection/ToolBar/index.js]: removed classnames dependency in [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/ebf4dd62097c9c4210f58d60bba43920842fbf4d). As clsx is already made available by material-ui, should we have to again make it as a dependency.

[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/ServiceCollection/ToolBar/index.js]: added clsx as explicit dependency [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/ebf4dd62097c9c4210f58d60bba43920842fbf4d)

File: src/components/AiMarketplace/MainSection/index.js [hunk 22]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from "react";
+import Grid from "@material-ui/core/Grid";
+import { withStyles } from "@material-ui/styles";
+
+import Filter from "./Filter";
+import ServiceCollection from "./ServiceCollection";
+import { useStyles } from "./styles";
+
+class MainSection extends Component {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/AiMarketplace/MainSection/index.js]: converted this as functional component. [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/ee809bbb8eb9eb0a55e848f47091f58e10ad0b0a)

File: src/components/Login/Signup/index.js [hunk 23]
```diff
@@ -0,0 +1,231 @@
...
+                        <StyledButton type="blue" btnText="Conitnue" onClick={this.handleConfirmSignup} />
+                    </div>
+                </form>
+            </Grid>
+        );
+
+        return (
+            <div>
+                <Grid container spacing={24} className={classes.signupMainContent}>
+                    {toBeConfirmed ? renderOTP : renderForm}
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Login/Signup/index.js]: separated them [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/69489b52c2330ebdf1969347d30a0c1effd12761)

File: src/components/Onboarding/Authentication/index.js [hunk 24]
```diff
@@ -0,0 +1,103 @@
...
+                <h3>Validate Email</h3>
+                <p className={classes.validateEmailDescription}>
+                    <span>A verification code has been sent to your registered email address.</span>
+                    <br /> <br />
+                    Please enter the verification code below to confirm your email address. If you are unable to find
+                    the email from <span>‘otp@singularitynet.io’</span> in your inbox, make sure to check the spam
+                    folder. The code will be valid for 5 minutes.{" "}
+                </p>
+                {loading ? <InlineLoader loading={loading} /> : null}
+                <TextField
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Onboarding/Authentication/index.js]: It is a controlled component ryt? It has
>  ```
>  value={verificationCode}
>  onChange={this.handleVerificationCode}
> ```

File: src/components/Onboarding/ProgressBar/ProgressSection/index.js [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+    ACTIVE: "active",
+    COMPLETED: "completed",
+};
+
+const ProgressSection = ({ progressNumber, progressText, progressStatus }) => {
+    const classes = useStyles();
+
+    return (
+        <li>
+            {progressStatus === ProgressStatusList.COMPLETED ? (
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Onboarding/ProgressBar/ProgressSection/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/f375f14c31fe9624b3ee93d429152aabc543cbe2)

File: src/components/Onboarding/ProgressBar/index.js [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+import ProgressSection, { ProgressStatusList } from "./ProgressSection";
+import { useStyles } from "./styles";
+
+const ProgressBar = ({ classes, activeSection }) => {
+    const computeProgressStatus = (progressNumber, activeSection) => {
+        if (progressNumber < activeSection) {
+            return ProgressStatusList.COMPLETED;
+        } else if (progressNumber === activeSection) {
+            return ProgressStatusList.ACTIVE;
+        } else if (progressNumber > activeSection) {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Onboarding/ProgressBar/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/fce9cf8d4818d831e38f30ca9e00c739ea00207f)

File: src/components/Onboarding/WalletKey/index.js [hunk 27]
```diff
@@ -0,0 +1,88 @@
...
+                            this.setState({ error: res.data, allowContinue: true });
+                            return;
+                        }
+                        this.setState({
+                            privateKey: res.data.data[0].private_key,
+                            allowContinue: true,
+                        });
+                    })
+                    .catch(err => {
+                        let error = parseError(err);
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Onboarding/WalletKey/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/f6ee530700a75ad4d5a68f9944c3cec6946e7ace)

File: src/components/Onboarding/index.js [hunk 28]
```diff
@@ -0,0 +1,124 @@
...
+    };
+
+    componentDidUpdate = () => {
+        console.log("mount route", this.props.isWalletAssigned);
+
+        if (this.props.isWalletAssigned) {
+            console.log("update route");
+            this.props.history.push(Routes.AI_MARKETPLACE);
+        }
+        if (sessionStorage.getItem(Session.USERNAME) && this.state.activeSection === 1) {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Onboarding/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/43f70939e592abc92e16617ee8ddb46f8040d5e0)

File: src/components/common/ErrorMsgBox/index.js [hunk 29]
```diff
@@ -0,0 +1,16 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "@material-ui/styles";
+
+import { useStyles } from "./styles";
+
+const ErrorMsgBox = ({ classes, showErr, errorMsg }) => {
+    return <Fragment> {showErr ? <p className={classes.errorText}>{errorMsg}</p> : null} </Fragment>;
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/common/ErrorMsgBox/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/0b1a09e5f7bc46aaa131bb4939c8d0e95e545ed7)

File: src/components/common/ErrorMsgText/index.js [hunk 30]
```diff
@@ -0,0 +1,14 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+import { useStyles } from "./styles";
+
+const ErrorMsgText = ({ error }) => {
+    const classes = useStyles();
+    return <Fragment>{error ? <p className={classes.errorMsg}>{error}</p> : null}</Fragment>;
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/common/ErrorMsgText/index.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/a6ec5b8aa1ef1758ea76c569cc1d95eeb92eb050)

File: src/Redux/actionCreators/UserActions.js [hunk 31]
```diff
@@ -0,0 +1,101 @@
...
+export const signOut = dispatch => {
+    let userDetails = {
+        type: SIGN_OUT,
+        payload: { isLoggedIn: true },
+    };
+    Auth.signOut()
+        .then(data => {
+            userDetails.payload.isLoggedIn = false;
+        })
+        .catch(err => {})
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/Redux/actionCreators/UserActions.js]: removed it [commit](https://github.com/singnet/refactored-dapp/pull/2/commits/9c2c6f00783543ecdeac19798a5afb8570910c61)

File: src/components/Login/Signup/index.js [hunk 32]
```diff
@@ -0,0 +1,231 @@
...
+        }
+        if (email === "") {
+            this.setState({ error: "Email cannot be left blank" });
+            return;
+        }
+        if (!isValidEmail(email)) {
+            this.setState({ error: "Please enter a valid email" });
+            return;
+        }
+        if (password === "") {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/Login/Signup/index.js]: will raise validate.js in separate PR

File: src/components/HOC/WithRegistrationHeader.js [hunk 33]
```diff
@@ -0,0 +1,18 @@
+import React, { Fragment, Component } from "react";
+
+import LoginOnboardingHeader from "../common/LoginOnboardingHeader";
+
+const withRegistrationHeader = (InputComponent, headerData) => {
+    return class extends Component {
```
[the remark this answers is not in the document]
Comment (Vivek205) [reply src/components/HOC/WithRegistrationHeader.js]: [done](https://github.com/singnet/refactored-dapp/pull/2/commits/7f44c50d631a978b1865208418fb424af44350de)

File: src/components/Onboarding/Authentication/index.js (hunk 24 above)
[the remark this answers is not in the document]
Comment (chetankothari) [reply src/components/Onboarding/Authentication/index.js]: Sorry, my bad, somehow missed the `value`.

**Pull request merged**